Docs/CommonErrors.md

```diff
@@ -0,0 +1,70 @@
+# Common Wallet Library Errors
+
+We return a Result object for every external interface that could have the potential to throw. Each public interface and what error could occur is outlined here.
+
+## Outer Errors
+### `VerifiedIdClient`
+
+`createVerifiedIdRequest`
+- `HttpError`:
+    - Code
+    - Retryable
+- `UnsupportedProtocolError`
+   - inner error with exact error
+   - Retryable: false
+
+`encode`
+- `UnsupportedVerifiedIdTypeError`
+    - Retryable: false
+
+`decode`
+- `InvalidRawDataError`
+    - Retyable: false
+
+### `VerifiedIdRequest`
+
+`complete`
+- `RequirementsNotMetError`, retryable: false
+- `HttpError`
+    - Code
+    - Retryable
+
+`cancel`
+- `MissingStateError`
+    - Retyable: false
+- `MissingCallbackURLError`
+    - Retyable: false
+- `HttpError`:
+    - Code
+    - Retryable
+   
+
+### `Requirement`
+
+`validate`
+- `RequirementNotMetError`
+- `VerifiedIdDoesNotMatchRequirementConstraintError`
```

Comment:
Oddly specific, making VerifiedIdRequirement first class. For context, when is validate() called by wallet library developers? If they're only checking for UI to highlight unfulfilled requirements, could this be an innerError for RequirementNotMetError?

Reply:
Interesting, on client side, we have not been doing inner error pattern. Are you suggesting doing inner error or just being a type of RequirementNotMetError?